File: Hotel/Repository.java
package Hotel;


import java.util.ArrayList;
import java.util.List;

public class Repository {
    private List<Habitacion> habitaciones = new ArrayList<>();
    private List<Reservas> reservas = new ArrayList<>();

    public void agregarHabitacion(Habitacion habitacion) {
        habitaciones.add(habitacion);
        System.out.println("Habitación " + habitacion.getNumero() + " añadida.");
    }

    public boolean reservarHabitacion(Cliente cliente, String numeroHabitacion) {
        for (Habitacion habitacion : habitaciones) {
            if (habitacion.getNumero().equals(numeroHabitacion) && habitacion.isDisponible()) {
                Reservas reserva = new Reservas(cliente, habitacion);
                reservas.add(reserva);
                habitacion.setDisponible(false);
                System.out.println("Reserva realizada para " + cliente.getNombre() + " en la habitación " + habitacion.getNumero() + ".");
                return true;
            }
        }
        System.out.println("Error: Habitación no disponible.");
        return false;
    }

    public void cancelarReserva(String numeroHabitacion) {
        Reservas reserva = null;
        for (Reservas r : reservas) {
            if (r.getHabitacion().getNumero().equals(numeroHabitacion)) {
                reserva = r;
                break;
            }
        }
        if (reserva != null) {
            reservas.remove(reserva);
            reserva.getHabitacion().setDisponible(true);
            System.out.println("Reserva cancelada para en la habitación " + numeroHabitacion + ".");
        } else {
            System.out.println("No se encontró la reserva en la habitación " + numeroHabitacion + ".");
        }
    }

    public void mostrarReservasActivas() {
        if (reservas.isEmpty()) {
            System.out.println("No hay reservas activas.");
        } else {
            System.out.println("Reservas activas:");
            for (Reservas reserva : reservas) {
                System.out.println("Se ha reservado la habitación " + reserva.getHabitacion().getNumero() + ".");
            }
        }
    }
}

File: Hotel/Repository.kt
package org.example.Hotel

class Repository {
    private val habitaciones: MutableList<Habitacion> = mutableListOf()
    private val reservas: MutableList<Reservas> = mutableListOf()

    fun agregarHabitacion(habitacion: Habitacion) {
        habitaciones.add(habitacion)
        println("Habitación ${habitacion.numero} añadida.")
    }

    fun reservarHabitacion(cliente: Cliente, numeroHabitacion: String): Boolean {
        val habitacion = habitaciones.find { it.numero == numeroHabitacion && it.disponible }
         if (habitacion != null) {
            val reserva = Reservas(cliente, habitacion)
            reservas.add(reserva)
            habitacion.disponible = false
            println("Reserva realizada para ${cliente.nombre} en la habitación ${habitacion.numero}.")
            return true
        } else {
            println("Error: Habitación no disponible.")
            return false
        }
    }

    fun cancelarReserva(numeroHabitacion: String) {
        val reserva = reservas.find { it.habitacion.numero == numeroHabitacion }
        if (reserva != null) {
            reservas.remove(reserva)
            reserva.habitacion.disponible = true
            println("Reserva cancelada para en la habitación ${numeroHabitacion}.")
        } else {
            println("No se encontró la reserva en la habitación ${numeroHabitacion}.")
        }
    }

    fun mostrarReservasActivas() {
        if (reservas.isEmpty()) {
            println("No hay reservas activas.")
        } else {
            println("Reservas activas:")
            for (numero in habitaciones) {
                println("Se ha reservado la habitación${numero.numero}.")
            }
        }
    }
}